packages/node/src/breadcrumbs/FileBreadcrumbsStorage.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    BacktraceAttachment,
+    Breadcrumb,
+    BreadcrumbLogLevel,
+    BreadcrumbType,
+    BreadcrumbsStorage,
+    RawBreadcrumb,
+    SessionFiles,
+    TimeHelper,
+    jsonEscaper,
+} from '@backtrace-labs/sdk-core';
+import path from 'path';
+import { BacktraceFileAttachment } from '../attachment';
+import { AlternatingFileWriter } from '../common/AlternatingFileWriter';
+
+const FILE_PREFIX = 'breadcrumbs';
+
+export class FileBreadcrumbsStorage implements BreadcrumbsStorage {
+    public get lastBreadcrumbId(): number {
+        return this._lastBreadcrumbId;
+    }
+
+    private _lastBreadcrumbId: number = TimeHelper.toTimestampInSec(TimeHelper.now());
+
+    private readonly _writer: AlternatingFileWriter;
+
+    constructor(private readonly _file1: string, private readonly _file2: string, maximumBreadcrumbs: number) {
+        this._writer = new AlternatingFileWriter(_file1, _file2, maximumBreadcrumbs);
+    }
+
+    public static createFromSession(session: SessionFiles): FileBreadcrumbsStorage | undefined {
+        const files = session
+            .getSessionFiles()
+            .filter((f) => path.basename(f).startsWith(FILE_PREFIX))
+            .slice(0, 2);
+
+        if (!files.length) {
```

Comment:
what does it mean? There is no breadcrumb file for the current session?  If it's unexpected, shouldn't be louder about that?

Reply:
It's not unexpected. There may be a possibility that the previous session did not create breadcrumbs at all.